src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,97 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_SIZE = 10;
+    private Object[] array;
+    private int size;
+    
+    public ArrayList() {
+        this.array = new Object[INITIAL_SIZE];
+    }
+    
     @Override
     public void add(T value) {
-
+        if (size == array.length - 1) {
+            increaseArray(array.length + array.length / 2 + 1);
+        }
+        array[size++] = value;
     }
-
+    
     @Override
     public void add(T value, int index) {
-
+        if (size != index) {
+            checkIndex(index);
+        }
+        if (size == array.length - 1) {
+            increaseArray(array.length + array.length / 2 + 1);
+        }
```

Comment:
```suggestion
        if (size == array.length) {
            increaseArray();
        }
            checkIndex(index);
```

Reply:
this condition just for add 
        if (size != index) {
            checkIndex(index);
        }
        because of  ArrayList structure has different condition of checking index. 
        It has  "rangeCheckForAdd"  where:  (index > size || index < 0)
        
        Maybe. It will better to create a new metod with similar condition?